### PR TITLE
fix CI can't pass, update test case

### PR DESCRIPTION
**Pull request by wangshub**
Hi @torrmal 

Found travis failed to run `python3 run_tests.py`, so I changed to `python setup.py install`. But the test still can not pass.

https://travis-ci.org/wangshub/mindsdb/jobs/499776388#L1132

image

### CODE REVIEW

**Comment by wangshub** [comment]
@torrmal I think i found the bug, i can fix it.

**Comment by torrmal** [comment]
Thank you! That will be fantastic

**Comment by wangshub** [comment]
@torrmal 

- [x] Fixed the test case bug under `python3.6` and `python3.7`
- [ ] bug under `python3.5` is Working in Progress

**Comment by wangshub** [comment]
f-Strings is not Compatible with `python3.5`, use str format instead 

@torrmal

**Comment by wangshub** [comment]
Aha! another bug appears under `python3.5` @torrmal 

```
INFO:mindsdb-logger-80ea6bfe-3b6d-11e9-a64a-0017fa02fdf3:libs/phases/model_trainer/model_trainer.py:74 - 'Trained: model test_one_label_prediction [OK], TOTAL TIME: 93.16 seconds'
INFO:mindsdb-logger-80ea6bfe-3b6d-11e9-a64a-0017fa02fdf3:libs/phases/base_module.py:79 - '[END] ModelTrainer, execution time: 93.160 seconds'
WARNING:mindsdb-logger-80ea6bfe-3b6d-11e9-a64a-0017fa02fdf3:libs/phases/base_module.py:60 - 'Target phase is different than PHASE_END, Only change this for debug purposes'
INFO:mindsdb-logger-80ea6bfe-3b6d-11e9-a64a-0017fa02fdf3:libs/phases/base_module.py:66 - '[START] ModelAnalyzer'
Traceback (most recent call last):
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 137, in _execute_learn
    self._call_phase_module('ModelAnalyzer')
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 94, in _call_phase_module
    return module()
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/phases/base_module.py", line 76, in __call__
    ret = self.run(**kwargs)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/phases/model_analyzer/model_analyzer.py", line 58, in run
    real_value=real_val, predicted_value=predicted_val, histogram=self.transaction.persistent_model_metadata.column_stats[pcol]['histogram'])
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/ml_models/probabilistic_validator.py", line 80, in register_observation
    predicted_value_b = self._get_value_bucket(predicted_value, histogram)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/ml_models/probabilistic_validator.py", line 66, in _get_value_bucket
    i = self._closest(histogram['x'], value)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/ml_models/probabilistic_validator.py", line 57, in _closest
    aux.append(abs(value-ele))
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'
ERROR:mindsdb-logger-80ea6bfe-3b6d-11e9-a64a-0017fa02fdf3:libs/controllers/transaction.py:146 - 'None'
ERROR:mindsdb-logger-80ea6bfe-3b6d-11e9-a64a-0017fa02fdf3:libs/controllers/transaction.py:147 - "unsupported operand type(s) for -: 'NoneType' and 'float'"
Traceback (most recent call last):
  File "run_tests.py", line 113, in <module>
    run_all_test_that_should_work()
  File "run_tests.py", line 110, in run_all_test_that_should_work
    test_one_label_prediction()
  File "run_tests.py", line 67, in test_one_label_prediction
    to_predict=label_name
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/predictor.py", line 168, in learn
    Transaction(session=self, transaction_metadata=transaction_metadata, logger=self.log, breakpoint=breakpoint)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 60, in __init__
    self.run()
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 215, in run
    self._execute_learn()
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 148, in _execute_learn
    raise e
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 137, in _execute_learn
    self._call_phase_module('ModelAnalyzer')
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/controllers/transaction.py", line 94, in _call_phase_module
    return module()
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/phases/base_module.py", line 76, in __call__
    ret = self.run(**kwargs)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/phases/model_analyzer/model_analyzer.py", line 58, in run
    real_value=real_val, predicted_value=predicted_val, histogram=self.transaction.persistent_model_metadata.column_stats[pcol]['histogram'])
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/ml_models/probabilistic_validator.py", line 80, in register_observation
    predicted_value_b = self._get_value_bucket(predicted_value, histogram)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/ml_models/probabilistic_validator.py", line 66, in _get_value_bucket
    i = self._closest(histogram['x'], value)
  File "/data/home/tesla/wanghub/mindsdb/venv/lib/python3.5/site-packages/mindsdb-1.0-py3.5.egg/mindsdb/libs/ml_models/probabilistic_validator.py", line 57, in _closest
    aux.append(abs(value-ele))
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'
```

**Comment by wangshub** [comment]
If mindsdb is compatible with `python3.5` ? @torrmal  😭

**Comment by wangshub** [comment]
Found the same problem 
[https://github.com/mindsdb/mindsdb/issues/76](https://github.com/mindsdb/mindsdb/issues/76)

**Comment by wangshub** [comment]
@ZoranPandovski @torrmal

**Comment by ZoranPandovski** [comment]
@wangshub  The main problem here is that the error in #76  is not related just to Python 3.5. Sometimes it appears also in 3.6 and 3.7 version

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,8 @@ notifications:
 cache: pip
 # install dependencies
 install:
-  - travis_wait pip install mindsdb
+  - pip install -r requirements.txt --quiet
```
Comment (ZoranPandovski) [inline .travis.yml]: Thanks @wangshub . I think We should keep this and install mindsdb from pip. Also, travis_wait is important so Travis will not throw timeouts when installing large requirements as torch, numpy etc

Comment (wangshub) [reply .travis.yml]: 1. we should keep pip install from local to test newest version
2. travis_wait is added
@ZoranPandovski

**Comment by wangshub** [comment]
> Great job @wangshub. Please just keep the Mindsdb travis install from pip.
> 
> > * travis_wait pip install mindsdb

@ZoranPandovski , OK, Change back to `travis_wait pip install mindsdb`

**Comment by wangshub** [comment]
But the ci won't pass, because pip doesn't install the newest mindsdb from repo. @ZoranPandovski

**Comment by torrmal** [comment]
I’ll push the latest

**Comment by torrmal** [comment]
@ZoranPandovski how can we fix the issue of unit tests using the latest pypi, so perhaps we should be able to for branches that are not master install it using the install.sh, as opposed to using pip, thoughts?

**Comment by ZoranPandovski** [comment]
@torrmal Our idea was to test successful installation from pip. Maybe we can add a different test for that, and for regression tests install latest mindsdb.

**Comment by torrmal** [comment]
Yes you are right

**Pull request closed without merging**